### PR TITLE
Event filters

### LINKED ISSUE #2528: [FEATURE] Filter events for virtual / in person
### Description

After this PR is merged https://github.com/EddieHubCommunity/LinkFree/pull/2527

Click the key cards to filter events

Screenshot 2022-12-28 at 22 09 27

### Screenshots

_No response_

### Additional information

_No response_

### LINKED ISSUE #2677: [FEATURE] Event CFP open icon and end date
### Description

Modify the events json to accept a line item for CFP end date with the following logic (open to discussion if we don't like the proposed logic or there is a better idea)

If CFP end date exists, populate CFP icon -- once the CFP date is passed this icon should automatically be removed
if CFP is left blank, do nothing (assumption being this is unknown or not an event that has a CFP)

Top legend should hold a new icon and explanation for CFP open such as:
Call for Papers Open
This event is accepting applications to speak.

Potential icons:
TbMicrophone2 
VscMegaphone

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by eddiejaoude**
## Fixes Issue

fixes #2528 
fixes #2677

Screenshot 2023-01-05 at 08 35 53

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2687"></a>

### CODE REVIEW

**Review by amandamartin-dev — APPROVED** [review]
LGTM but would suggest one more reviewer that works in the project more often than I do 

One edit you can consider: Defining CFP on the legend

Thoughts about possible follow up issues needed due to this update:
Is the mobile view too stacked with the legend now that there are 4 items? Possible need for a dropdown or some other type of filter on mobile?
This will require an update to documentation on the events directions likely as well

**Review by dan-mba — CHANGES_REQUESTED** [review]
You didn't update  `EventKey` with the new parameters [here](https://github.com/EddieHubCommunity/LinkFree/blob/283e698b198e85db433f6a2291ebe044c353a989/components/user/UserEvents.js#L8)

File: pages/events.js [hunk 1]
```diff
@@ -20,6 +21,18 @@ export async function getServerSideProps(context) {
 }
 
 export default function Events({ events }) {
+  const [eventType, seteventType] = useState("all");
+  let categorisedEvents = {
+    all: events,
+    virtual: events.filter((event) => event.isVirtual === true),
+    inPerson: events.filter((event) => event.isInPerson === true),
+    cfpOpen: events.filter(
+      (event) => new Date(event.date.cfpClose) > new Date()
```
Comment (dan-mba) [inline pages/events.js:30]: Is `date.cfpClose` a required field?

Comment (eddiejaoude) [reply pages/events.js:30]: Good spot, it is not required, so I should make a check 👍

**Comment by eddiejaoude** [comment]
Thanks @amandamartin-dev , comments inline below ...

> One edit you can consider: Defining CFP on the legend

Good point 👍 

> Thoughts about possible follow up issues needed due to this update:
> Is the mobile view too stacked with the legend now that there are 4 items? Possible need for a dropdown or some other type of filter on mobile?

Yep, this is true 👍  hopefully it can come in another PR soon.

> This will require an update to documentation on the events directions likely as well

I think I have covered that, but if not let me know

**Comment by eddiejaoude** [comment]
> You didn't update  EventKey with the new parameters [here](https://github.com/EddieHubCommunity/LinkFree/blob/283e698b198e85db433f6a2291ebe044c353a989/components/user/UserEvents.js#L8)

Thanks @dan-mba , great spot 👍 - should be fixed now

**Review by dan-mba — CHANGES_REQUESTED** [review]
The changes look good.
Just waiting on the cfpOpen filter check in pages/events.js

**Comment by eddiejaoude** [comment]
> Just waiting on the cfpOpen filter check in pages/events.js

OMG I missed that 🤦‍♂️ thank you Dan 🔥 I will do that now

**Review by dan-mba — APPROVED** [review]
Looks good

**Pull request merged**